### PR TITLE
Enable lowercase tree selector

### PR DESCRIPTION
**Pull request by nikolasleblanc**
Codelyzer doesn't like title case selectors: https://github.com/mgechev/codelyzer

> "component-selector": [true, "element", "sg", "kebab-case"]

This allows Tree and tree to be used as selectors

### CODE REVIEW

**Comment by sjdemoor** [comment]
👍

File: lib/components/tree.component.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { KEYS } from '../constants/keys';
 import * as _ from 'lodash';
 
 @Component({
-  selector: 'Tree',
+  selector: 'Tree, tree',
```
Comment (adamkleingit) [inline lib/components/tree.component.ts]: Hey,
I prefer to allow using kebab-case names, but issue a "deprecated" warning (using lib/deprecated) if someone is using the Capitalized selectors.
If you have time to update the PR that would be nice.

Comment (nikolasleblanc) [inline lib/components/tree.component.ts]: How would you discern one selector's usage from another? If I try via the dom (el.nativeElement.localName), I get auto-lowercased values, and if I use reflect (http://stackoverflow.com/questions/37188216/accessing-selector-from-within-an-angular-2-component), I'll run into an AoT issue (https://github.com/angular/angular/issues/13495#issuecomment-267314384). 

If you've got any tips or tricks I'd be happy to hear them.

Comment (adamkleingit) [inline lib/components/tree.component.ts]: Let's just add a warning message in the constructor that says: "if you are using the UpperCase selector please move to the kebab-case selector, as the UpperCase will be soon deprecated".

Can you also add kebab-case selectors to all other directives and components of the library?

Thanks for the help!

**Comment by nikolasleblanc** [comment]
Rebased against your new changes after setting up new selectors and a deprecatedSelector function for outputting the console.warn, please double check to ensure I haven't borked anything.

**Comment by adamkleingit** [comment]
Merged.
I passed the elementRef and checked `elementRef.nativeElement.tagName`

**Comment by nikolasleblanc** [comment]
On dev tools that always returned lowercase tag names no matter what the
selector, you saw different results?

**Pull request merged**